### PR TITLE
Add access permissions check for >= 3.0

### PR DESCRIPTION
**Pull request by nadjaheitmann**
Add access permission test to CI and fix permission for the test to succeed, base for #154

### CODE REVIEW

File: lib/foreman_puppet/register.rb [hunk 1]
```diff
@@ -59,89 +59,89 @@
...
+                                                      lookup_values: [:destroy],
+                                                      'foreman_puppet/api/v2/smart_class_parameters': [:destroy],
+                                                      'foreman_puppet/api/v2/override_values': %i[create update destroy] },
       resource_type: 'ForemanPuppet::PuppetclassLookupKey'
   end
 
   security_block :puppet_environments do
-    permission :view_environments, { 'foreman_puppet/environments': %i[index show auto_complete_search welcome],
-                                     'foreman_puppet/api/v2/environments': %i[index show] },
+    permission :puppet_view_environments, { 'foreman_puppet/environments': %i[index show auto_complete_search welcome],
```
Comment (timogoebel) [inline lib/foreman_puppet/register.rb:101]: Do we need db migrations to change the permission names?

Comment (timogoebel) [reply lib/foreman_puppet/register.rb:101]: On second thought, do we want to change the name here? I'm wondering if the `puppet` prefix breaks things or not.

Comment (nadjaheitmann) [reply lib/foreman_puppet/register.rb:101]: Permissions test for master:
https://github.com/ATIX-AG/foreman_puppet/runs/2882491673?check_suite_focus=true

To me it looks like there is a name clash between original permissions and the plugin and the original permissions are overwritten: https://github.com/theforeman/foreman/blob/3f7804fd96d380010b340f2a954f9cb42a60c551/app/registries/foreman/plugin.rb#L330 
That is why I tried renaming. Correct me if I am wrong.

Comment (nadjaheitmann) [reply lib/foreman_puppet/register.rb:101]: But maybe overwriting the existing permission is exactly what we want and we should let the tests fail for Foreman < 3.0 ?

Comment (timogoebel) [reply lib/foreman_puppet/register.rb:101]: Yep, sounds good. With the plugin we do want to override core functionality, so I'd be ok with that.

Comment (nadjaheitmann) [reply lib/foreman_puppet/register.rb]: I have changed the PR to contain the permissions check starting from a 3.X stable branch.

File: .github/workflows/ruby_tests.yml [hunk 2]
```diff
@@ -123,6 +123,10 @@ jobs:
         if: github.event_name != 'push'
         run: |
           bundle exec rake test:foreman_puppet
+      - name: Run access permission test
+        if: ${{ startsWith(matrix.foreman-core-branch, '3') && github.event_name != 'push' }}
```
Comment (timogoebel) [inline .github/workflows/ruby_tests.yml:127]: Is this code currently reachable?

Comment (nadjaheitmann) [reply .github/workflows/ruby_tests.yml:127]: The if condition is being checked, but the test does not run as we do not have a foreman core branch starting with '3' (, yet).

File: .github/workflows/ruby_tests.yml [hunk 3]
```diff
@@ -123,6 +123,10 @@ jobs:
...
+        run: |
+          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
```
Comment (ezr-ondrej) [inline .github/workflows/ruby_tests.yml:129]: btw we don't need to test all the routes, we can just test the engine routes like:
https://github.com/theforeman/foreman_statistics/blob/master/test/unit/foreman_statistics/access_permissions_test.rb
OR
https://github.com/theforeman/foreman_bootdisk/blob/master/test/unit/access_permissions_test.rb

Comment (nadjaheitmann) [reply .github/workflows/ruby_tests.yml:137]: There already is https://github.com/theforeman/foreman_puppet/blob/master/test/unit/foreman_puppet/access_permissions_test.rb

Comment (nadjaheitmann) [reply .github/workflows/ruby_tests.yml:137]: Shall we close the PR if the test is already there?

Comment (ezr-ondrej) [reply .github/workflows/ruby_tests.yml:137]: this effort shown that the test doesn't test the integration with core routes.

So however we decide to rewrite core API routes (eighter directly in core, as I have that in my PR or with your PR here) that would be untested. So this test still brings value in testing that override.

Comment (nadjaheitmann) [reply .github/workflows/ruby_tests.yml:137]: Right. Shall we move ahead and merge it, then?

**Pull request merged**